vmm/src/config.rs

```diff
@@ -2130,22 +2143,71 @@ impl NumaConfig {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize, Default)]
+pub struct RestoredNetConfig {
+    pub id: String,
+    #[serde(default)]
+    pub num_fds: usize,
+    #[serde(
+        default,
+        serialize_with = "serialize_restorednetconfig_fds",
+        deserialize_with = "deserialize_restorednetconfig_fds"
+    )]
+    pub fds: Option<Vec<i32>>,
+}
+
+fn serialize_restorednetconfig_fds<S>(
+    x: &Option<Vec<i32>>,
+    s: S,
+) -> std::result::Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    if let Some(x) = x {
+        warn!("'RestoredNetConfig' contains FDs that can't be serialized correctly. Serializing them as invalid FDs.");
+        let invalid_fds = vec![-1; x.len()];
+        s.serialize_some(&invalid_fds)
+    } else {
+        s.serialize_none()
+    }
+}
+
+fn deserialize_restorednetconfig_fds<'de, D>(
+    d: D,
+) -> std::result::Result<Option<Vec<i32>>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let invalid_fds: Option<Vec<i32>> = Option::deserialize(d)?;
+    if let Some(invalid_fds) = invalid_fds {
+        warn!("'RestoredNetConfig' contains FDs that can't be deserialized correctly. Deserializing them as invalid FDs.");
+        Ok(Some(vec![-1; invalid_fds.len()]))
+    } else {
+        Ok(None)
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize, Default)]
 pub struct RestoreConfig {
     pub source_url: PathBuf,
     #[serde(default)]
     pub prefault: bool,
+    #[serde(default)]
+    pub net_fds: Option<Vec<RestoredNetConfig>>,
 }
 
 impl RestoreConfig {
     pub const SYNTAX: &'static str = "Restore from a VM snapshot. \
-        \nRestore parameters \"source_url=<source_url>,prefault=on|off\" \
+        \nRestore parameters \"source_url=<source_url>,prefault=on|off\", \
+        net_fds=<list_of_net_ids_with_their_associated_fds> \
```

Comment:
One nit that I missed it in the previous review: the closing `"` should be in the end, e.g. 

```
        \nRestore parameters \"source_url=<source_url>,prefault=on|off, \
        net_fds=<list_of_net_ids_with_their_associated_fds>\" \
```

Reply:
My bad. Updated it.